nostr/relay.py

```diff
@@ -54,42 +51,55 @@ def __post_init__(self):
             on_error=self._on_error,
             on_close=self._on_close
         )
+        self._connection_thread: Thread = None
 
-    def connect(self):
-        self.ws.run_forever(
-            sslopt=self.ssl_options,
-            http_proxy_host=self.proxy_config.host if self.proxy_config is not None else None, 
-            http_proxy_port=self.proxy_config.port if self.proxy_config is not None else None,
-            proxy_type=self.proxy_config.type if self.proxy_config is not None else None,
-        )
+    def connect(self, is_reconnect=False):
+        if not self.is_connected():
+            with self.lock:
+                self._connection_thread = Thread(
+                    target=self.ws.run_forever,
+                    kwargs={
+                        "sslopt": self.ssl_options,
+                        "http_proxy_host": self.proxy_config.host if self.proxy_config is not None else None,
+                        "http_proxy_port": self.proxy_config.port if self.proxy_config is not None else None,
+                        "proxy_type": self.proxy_config.type if self.proxy_config is not None else None
+                    },
+                    name=f"{self.url}-connection"
+                )
+                self._connection_thread.start()
+
+            if not is_reconnect:
+                Thread(
+                    target=self.outgoing_messages_worker,
+                    name=f"{self.url}-outgoing-messages-worker",
+                    daemon=True
+                ).start()
 
+            time.sleep(1)
```

Comment:
Why do we sleep here? Shouldn't be necessary with the queue and the connection state anymore.